extensions/godslayerakp/http.js

```diff
@@ -0,0 +1,445 @@
+    };
+    // @ts-ignore
+    const oldPrepare = extensionManager._prepareBlockInfo.bind(extensionManager);
+    // @ts-ignore
+    extensionManager._prepareBlockInfo = (serviceName, blockInfo) => {
+        if (typeof blockInfo === 'string' && blockInfo.startsWith('---'))
+            return blockInfo;
+        return oldPrepare(serviceName, blockInfo);
+    };
+
+    // the funny class to make event blocks look better
+    class Events {
+        constructor() {
+            this.events = {};
+        }
+
+        add(name) {
+            this.events[name] = false;
+        }
+
+        peek(name) {
+            return this.events[name];
+        }
+
+        check(name) {
+            const state = this.events[name];
+            this.events[name] = false;
+            return state;
+        }
+
+        activate(name) {
+            this.events[name] = true;
+        }
+    }
+
+    /* ------- BLOCKS -------- */
+    const {BlockType, Cast, ArgumentType} = Scratch;
+
+    class WebRequests {
+        static get defaultRequest() {
+            const defaultRequest = {
+                events: new Events(),
+                get mimeType() {
+                    return this.options.headers['Content-Type'];
+                },
+                set mimeType(value) {
+                    this.options.headers['Content-Type'] = value;
+                },
+                set method(val) {
+                    this.options.method = val;
+                    // remove body on get requests
+                    if (val === 'GET') {
+                        delete this.options.body;
+                    }
+                },
+                get method() {
+                    return this.options.method;
+                },
+                options: {
+                    headers: {
+                        'Content-Type': 'text/plain'
+                    },
+                    method: 'GET'
+                },
+                set body(val) {
+                    if (this.method === 'GET') return;
+                    this.options.body = val;
+                },
+                get body() {
+                    return this.options.body;
+                }
+            };
+
+            defaultRequest.events.add('reqEnd');
+            defaultRequest.events.add('reqSuccess');
+            defaultRequest.events.add('reqFail');
+
+            return defaultRequest;
+        }
+        static get defualtResponse() {
+            const defualtResponse = {
+                text: '',
+                status: '',
+                statusText: '',
+                headers: new Headers(),
+                error: '',
+                url: ''
+            };
+
+            return defualtResponse;
+        }
+
+        /**
+         * no need to install runtime as it comes with Scratch var
+         */
+        constructor() {
+            this.clearAll();
+        }
+        getInfo() {
+            return {
+                id: 'extawebrequestsHTTP',

```

Comment:
prefix extension ids with something related to username

Reply:
oh

Reply:
fixed...